spec/services/invoices/payments/adyen_service_spec.rb

```diff
@@ -0,0 +1,244 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Invoices::Payments::AdyenService, type: :service do
+  subject(:adyen_service) { described_class.new(invoice) }
+
+  let(:customer) { create(:customer) }
+  let(:organization) { customer.organization }
+  let(:adyen_payment_provider) { create(:adyen_provider, organization:) }
+  let(:adyen_customer) { create(:adyen_customer, customer:) }
+  let(:adyen_client) { instance_double(Adyen::Client) }
+  let(:payments_api) { Adyen::PaymentsApi.new(adyen_client, 70) }
+  let(:checkout) { Adyen::Checkout.new(adyen_client, 70) }
+  let(:payments_response) { generate(:adyen_payments_response) }
+  let(:payment_methods_response) { generate(:adyen_payment_methods_response) }
+
+  let(:invoice) do
+    create(
+      :invoice,
+      organization:,
+      customer:,
+      total_amount_cents: 1000,
+      currency: 'USD',
+      ready_for_payment_processing: true
+    )
+  end
+
+  describe '#create' do
+    before do
+      adyen_payment_provider
+      adyen_customer
+
+      allow(Adyen::Client).to receive(:new)
+        .and_return(adyen_client)
+      allow(adyen_client).to receive(:checkout)
+        .and_return(checkout)
+      allow(checkout).to receive(:payments_api)
+        .and_return(payments_api)
+      allow(payments_api).to receive(:payments)
+        .and_return(payments_response)
+      allow(payments_api).to receive(:payment_methods)
+        .and_return(payment_methods_response)
+      allow(Invoices::PrepaidCreditJob).to receive(:perform_later)
+    end
+
+    fit 'creates an adyen payment' do
+      result = adyen_service.create
+
+      expect(result).to be_success
+
+      aggregate_failures do
+        expect(result.invoice).to be_succeeded
+        expect(result.invoice.payment_attempts).to eq(1)
+        expect(result.invoice.reload.ready_for_payment_processing).to eq(false)
+
+        expect(result.payment.id).to be_present
+        expect(result.payment.invoice).to eq(invoice)
+        expect(result.payment.payment_provider).to eq(adyen_payment_provider)
+        expect(result.payment.payment_provider_customer).to eq(adyen_customer)
+        expect(result.payment.amount_cents).to eq(invoice.total_amount_cents)
+        expect(result.payment.amount_currency).to eq(invoice.currency)
+        expect(result.payment.status).to eq('Authorised')
+
+        expect(adyen_customer.reload.payment_method_id).
```

Comment:
Layout/DotPosition: Place the . on the next line, together with the method name. (https://rubystyle.guide#consistent-multi-line-chains)